src/components/layout/Shell.tsx

```diff
@@ -82,7 +82,13 @@ export function Shell(props: Props) {
                 <SiteLogo />
               </Link>
 
-              <SidebarLinks onLinkClick={closeMobileNav} />
+              <SidebarLinks
+                onLinkClick={(link) => {
+                  if (!link.children) {
```

Comment:
Only close the mobile menu if it's not a parent link (i.e. it doesn't have children)